lib/node/rpc.js

```diff
@@ -2346,10 +2347,99 @@ class RPC extends RPCBase {
     return result;
   }
 
+  async deriveAddresses(args, help) {
+    if (help || args.length > 2 || args.length === 0)
+      throw new RPCError(errs.MISC_ERROR, 'deriveaddresses "descriptor"');
+
+    const valid = new Validator(args);
+
+    const desc = parseDescriptor(valid.str(0, ''), this.network, true);
+
+    if (desc.isRange() && args.length === 1) {
+      throw new RPCError(
+        errs.INVALID_PARAMETER,
+        'Range must be specified for ranged descriptor'
+      );
+    }
+
+    if (!desc.isRange() && args.length > 1) {
+      throw new RPCError(
+        errs.INVALID_PARAMETER,
+        'Range should not be specified for un-ranged descriptor'
+      );
+    }
+
+    let low = 0, high = 0;
+
+    try {
+      high = valid.u32(1, 0);
+    } catch (e) {
+      try {
+        const arr = valid.array(1, []);
+        low = arr[0];
+        high = arr[1];
+      } catch (e) {
+        throw new RPCError(
+          errs.INVALID_PARAMETER,
+          'Invalid range.'
+        );
+      }
+    }
+
+    this.isValidRange(low, high);
```

Comment:
Maybe we should rename `isValidRange()` to `getRangeParameters()` and move this code inside `getRangeParameters()`. 
Do something like 
```js
const { low, high } = getRangeParameters(args);
```

Reply:
Done.

Reply:
resolved